roles/dotfiles/tasks/main.yaml

```diff
@@ -1,3 +1,12 @@
 ---
+- name: temporarily downloading your dotfiles
+  git: src={{dotfiles_repo}} dest=/tmp/ansibox.dotfiles
+  when: "{{ dotfiles_repo | default(false)}}"
+
 - name: sync dotfiles
-  synchronize: src=dotfiles/ dest=~/
+  synchronize: src=/tmp/ansibox.dotfiles/ dest=~/ --exclude=.git
+  when: "{{ dotfiles_repo | default(false)}}"
+
+- name: clear your temporary dotfiles download
+  file: path=/tmp/ansibox.dotfiles state=absent
```

Comment:
This is a bad pattern. using a predictable path in /tmp that just about anybody can write to, you can wind up deleting something you don't want to. Probably best to stash the clone somewhere in the working directory that's been gitignored, and that way it doesn't have to re-clone every time, it could be a persistent clone within the project tree.